engine/routes.py

```diff
@@ -159,6 +169,35 @@ def estimate():
     return jsonify(util.calc.get_estimate(input_data, request.url))
 
 
+@app.route('/stats', methods=['POST'])
+@set_timeout()
+def stats():
+    """
+    Return the calculated download size and completion time for a completed request by looking at the output directory
+    """
+    input_data = request.get_json()
+
+    try:
+        if app.config['AGGREGATE']:
+            async_job_dir = get_final_dir(input_data)
+        else:
+            async_job_dir = get_local_dir(input_data)
+
+        if os.path.isdir(async_job_dir):
+            return jsonify(util.calc.get_stats(async_job_dir))
+        else:
+            raise InvalidPathException("Directory %r does not exist" % async_job_dir)
+    except InvalidPathException as e:
+        output = {
+            "code": 500,
+            "message": "Request for download size and completion time failed for the following reason: %s" % e.message,
+            'requestUUID': input_data.get('requestUUID', '')
+        }
+        json_str = json.dumps(output, indent=2, separators=(',', ': '))
+        log.error(json_str)
+        return Response(json_str, mimetype='application/json')
```

Comment:
I'm confused. It seems like `InvalidPathException` is only raised manually in the `else` case. If so, the `except` block should just be moved to the `else` block.

Reply:
Both get_local_dir and get_final_dir can raise InvalidPathExceptions.